### PR TITLE
docs: Adds connectors' documentation

### PR DESCRIPTION
**Pull request by fran-unstructured**
Solves issue #653 by adding a connectors section to https://unstructured-io.github.io/unstructured/

### CODE REVIEW

File: docs/source/connectors.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+.. code:: shell
+
+  unstructured-ingest \
+    --remote-url s3://utic-dev-tech-fixtures/small-pdf-set/ \
+    --s3-anonymous \
+    --structured-output-dir s3-small-batch-output \
+    --num-processes 2
+
+To run this example, you'll first need to install the S3 dependencies as shown `here <https://unstructured-io.github.io/unstructured/connectors.html#s3-connector>`_.
```
Comment (ryannikolaidis) [inline docs/source/connectors.rst:17]: double check that link. it 404'd for me

File: docs/source/connectors.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+
+``Elasticsearch Connector``
+----------------------------
+You can preprocess documents stored in Elasticsearch by using the `Elasticsearch Connector <https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/connector/elasticsearch.py>`_. You can find an example of how to use it `here <https://github.com/Unstructured-IO/unstructured/blob/f5541c7b0b1e2fc47ec88da5e02080d60e1441e2/examples/ingest/elasticsearch/ingest.sh>`_.
+
+To install all dependencies for this connector run: ``pip install unstructured[elasticsearch]``
+
+
+``GCS Connector``
```
Comment (ryannikolaidis) [inline docs/source/connectors.rst:55]: Probably write out Google Cloud Storage?

**Review by ryannikolaidis — CHANGES_REQUESTED** [review]
A couple small notes, otherwise looks good to me!

File: docs/source/connectors.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
+Connectors
+==========
+Connect your preprocessing pipeline with your favorite data storage platforms, and batch process all your documents using the provided CLI to store structured outputs locally on your filesystem. 
+
+To use any connector you'll need to first install ``local-inference``. You can find more information about this in the `installation guide <https://unstructured-io.github.io/unstructured/installing.html>`_.
```
Comment (MthwRobinson) [inline docs/source/connectors.rst:5]: Small adjustment on this, you only need the `local-inference` extra if you're processing PDFs or images. However, you _do_ need the extra associated with the connector you're trying to us. So you'd need `pip install "unstructured[s3]"` or `pip install "unstructured[gcs,local-inference]"` if you have PDFs/images and want to use the GCS connector.

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
Thanks for this! In addition to @ryannikolaidis 's comments, just one small note on installation.

**Review by MthwRobinson — APPROVED** [review]
Looks good from my end with the updates

File: docs/source/connectors.rst (hunk 1 above)
Comment (MthwRobinson) [reply docs/source/connectors.rst:15]: I think you're good to leave that in here

Comment (ryannikolaidis) [reply docs/source/connectors.rst:15]: ahh right 🤦 was looking at this too late at night. yes, totally

**Review by ryannikolaidis — APPROVED** [review]
lgtm! nice work!

**Pull request merged**